### ISSUE TITLE
DRAM auto-slice offers only height/width slicing, so a deep but spatially tiny depthwise conv has no usable slice axis and aborts

### ISSUE THREAD

**Issue by sonalibaskaran2499** [tenstorrent/tt-metal#54564]
## Description

`ttnn::conv2d` / `ttnn::conv1d` with a DRAM input abort when the op does not fit L1 and neither spatial dimension can be split. `Op2DSliceConfig::SliceType` offers only `DRAM_HEIGHT` and `DRAM_WIDTH`, both spatial, so `determine_slice_config_internal` tries width, falls back to height, and then `TT_FATAL`s even when the channel dimension is large and could be split.

This is reached by the Gated DeltaNet short convolution in Qwen 3.5 / Qwen 3.6 35B-A3B, a depthwise conv over the fused QKV stream where `conv_dim = linear_key_head_dim * linear_num_key_heads * 2 + linear_value_head_dim * linear_num_value_heads = 8192`. The output is `1 x 23 x 8192`: height is 1, so it yields one slice, and width 23 rounds into a single 32-wide tile, so it also yields one slice. Meanwhile the dimension actually consuming L1 - 8192 channels  is not a slicing candidate at all, which is why the message reports "up to 1 slices".

For a depthwise conv there is no cross-channel reduction, so splitting channels is trivially legal and needs no halo. Splitting into 2 chunks of 4096 already brings the estimated L1 use from 2523200 B to 1261632 B against 1428608 B available.

The gap is general rather than specific to this model: any op reaching `op_slicing` whose output is spatially small but deep is unsliceable today. The channel axis is legal for most of them  output channels are never reduced over in a convolution, and pooling is fully per-channel.

## Parameters that trigger the overflow

```
nn.Conv1d(in_channels=8192, out_channels=8192, kernel_size=4, groups=8192,
padding=3, bias=False)   # input length 20 -> output 1 x 23 x 8192
```

## Steps to reproduce

```python
import torch
import ttnn

C, K, L = 8192, 4, 20          # Qwen 3.5 / 3.6 35B-A3B gated-delta short conv
PAD = K - 1                    # causal padding -> output 1 x 23 x 8192

torch.manual_seed(0)
device = ttnn.open_device(device_id=0, l1_small_size=32768)
try:
    x = torch.randn(1, L, C, dtype=torch.bfloat16)
    w = torch.randn(C, 1, 1, K, dtype=torch.bfloat16)

    out, out_len = ttnn.conv1d(
        input_tensor=ttnn.from_torch(
            x.reshape(1, 1, -1, C), ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT,
            device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG,
        ),
        weight_tensor=ttnn.from_torch(w, ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT),
        device=device,
        in_channels=C, out_channels=C, batch_size=1, input_length=L,
        kernel_size=K, stride=1, padding=PAD, dilation=1,
        groups=C,                                        # depthwise
        return_output_dim=True,
    )
    print(f"ran: output length {out_len}")
finally:
    ttnn.close_device(device)
```

## Error

```
TT_FATAL @ ttnn/cpp/ttnn/operations/sliding_window/op_slicing/op_slicing.cpp:266: found_valid_config
info:
DRAM Auto slice could not find valid slice configuration. Tried up to 1 slices for
width-slicing on output dimension 23. Available L1: 1428608 bytes. Operation requires
more memory than available even with maximum slicing.
```

## Root Cause

`Op2DSliceConfig::SliceType` has no channel option, and the whole `OpSliceAttr` interface is spatially typed `IOShape` is a `(height, width)` pair, so slice bounds have no channel coordinate to travel in. `run_sliced_op` correspondingly hardcodes the full channel range in both `padded_slice` (`0 .. input_channels`) and `slice_write` (`0 .. output_channels`).

When both spatial extents are degenerate there is therefore no axis left, regardless of how much slack the channel dimension has.

**Comment by sonalibaskaran2499** [comment]
As it would take significant effort to make this a robust conv1d/2d feature I have implemented workaround in the model side. Therefore closing this issue.
